sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/dtt/DOTypeAdapter.java

```diff
@@ -317,65 +317,70 @@ public TPredefinedCDCEnum getCdc() {
      * return a list of Resumed Data Type Templates beginning from this DoType (Do or SDO).
      * @apiNote This method doesn't check relationship between DO/SDO and DA. Check should be done by caller
      * @param rootRDTT reference Resumed Data Type Template used to build the list
-     * @param visitedSdo a cache to stored visited SDO
      * @param filter filter for DO/SDO and DA/BDA
      * @return list of Resumed Data Type Templates beginning from this DoType (Do or SDO)
      */
-    public List<ResumedDataTemplate> getResumedDTTs(ResumedDataTemplate rootRDTT,
+    public List<ResumedDataTemplate> getResumedDTTs(ResumedDataTemplate rootRDTT, ResumedDataTemplate filter) {
+        return getResumedDTTs(rootRDTT, new HashSet<>(), filter);
+    }
+
+    private List<ResumedDataTemplate> getResumedDTTs(ResumedDataTemplate rootRDTT,
                                     Set<String> visitedSdo, ResumedDataTemplate filter) {
 
-        List<ResumedDataTemplate> rDtts = new ArrayList<>();
+        List<ResumedDataTemplate> resultRDTTs = new ArrayList<>();
         for(TUnNaming tUnNaming: currentElem.getSDOOrDA()){
             if(tUnNaming.getClass() == TDA.class){
                 TDA tda = (TDA)tUnNaming;
-                if(filter.isDaNameDefined() &&
-                        !filter.getDaName().getName().equals(tda.getName()) ){
-                    // filter out
-                    continue;
-                }
-
-                rootRDTT.getDaName().setName(tda.getName());
-                rootRDTT.getDaName().setFc(tda.getFc());
-                rootRDTT.getDaName().setBType(tda.getBType());
-                if(tda.getBType() == TPredefinedBasicTypeEnum.STRUCT){
-                    DATypeAdapter daTypeAdapter = parentAdapter.getDATypeAdapterById(tda.getType()).orElse(null);
-                    if(daTypeAdapter != null){
-                        // get list of Resumed Data Type Templates beginning from this DAType
-                        List<ResumedDataTemplate> resumedDataTemplateList = daTypeAdapter.getResumedDTTs(
-                                rootRDTT,new HashSet<>(), filter
-                        );
-                        rDtts.addAll(resumedDataTemplateList);
-                    }
-                } else {
-                    ResumedDataTemplate resumedDataTemplate = ResumedDataTemplate.copyFrom(rootRDTT);
-                    resumedDataTemplate.getDaName().setType(tda.getType());
-                    resumedDataTemplate.getDaName().setValImport(tda.isValImport());
-                    resumedDataTemplate.setDaiValues(tda.getVal());
-                    rDtts.add(resumedDataTemplate);
-                }
+                resultRDTTs.addAll(getDttOfDA(rootRDTT, filter, tda));
```

Comment:
getDttOfDA ... not very readable, avoid to use acronyms

Reply:
Is `getResumedDTTsOfDA` OK for you ? Otherwise what do you suggest ?